lark/tree.py

```diff
@@ -35,7 +35,10 @@ def __init__(self):
         self.empty = True
 
 
-class Tree(object):
+_Leaf_T = TypeVar("_Leaf_T")
+
+
+class Tree(Generic[_Leaf_T]):
```

Comment:
Are there any implication, in terms of speed, or memory consumption (or otherwise), of inheriting from Generic?

Reply:
Very minimal, if at all. Especially Python3.8 and up (I think) really minimize the impact.